src/config/config-filter.c

```diff
@@ -204,11 +212,18 @@ config_filter_find_all(struct config_filter_context *ctx, pool_t pool,
 	ARRAY_TYPE(config_filter_parsers) matches;
 	ARRAY_TYPE(const_string) service_names;
 	unsigned int i;
+	unsigned int t_id;
 
 	memset(output_r, 0, sizeof(*output_r));
 
 	p_array_init(&matches, pool, 8);
 	p_array_init(&service_names, pool, 8);
+
+	/* create a new stack after matches has been allocated
+	 * so we can free the memory used by config_filter_match_rest
+	 * with the below t_pop call */
+	t_id = t_push_named("config_filter_find_all");
```

Comment:
Well, I suppose this goes as well, but I ment literally T_PUSH { config_filter_find_all() } T_POP;